shuup/core/api/basket.py

```diff
@@ -506,12 +521,35 @@ def set_shipping_address(self, request, *args, **kwargs):
             return Response(data, status=status.HTTP_200_OK)
 
     @detail_route(methods=['post'])
+    def set_shipping_method(self, request, *args, **kwargs):
+        return self._handle_setting_method(request, ShippingMethod, "shipping_method")
+
+    @detail_route(methods=['post'])
+    def set_payment_method(self, request, *args, **kwargs):
+        return self._handle_setting_method(request, PaymentMethod, "payment_method")
+
+    def _handle_setting_method(self, request, model, attr_field):
+        self.process_request()
+        serializer = MethodIDSerializer(data=request.data)
+        if serializer.is_valid():
```

Comment:
you can invoke `serializer.is_valid(raise_exception=True)` so you don't need to return `return Response(serializers.errors, status=status.HTTP_400_BAD_REQUEST)` explicitly and as a bonus we remove a indentation :)